### PR TITLE
Offline optimizer

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

File: platform/lib/build/pageMinifier.js [hunk 1]
```diff
@@ -87,22 +90,39 @@ class PageMinifier {
...
           callback();
         }))
         .pipe(gulp.dest('./'));
   }
 
+
+  async optimize(html, path) {
+    const ampRuntimeVersion = await runtimeVersionPromise;
+    return ampOptimizer.transformHtml(html, {
+      ampUrl: path.replace(/^pages/, ''),
```
Comment (sebastianbenz) [inline platform/lib/build/pageMinifier.js]: @matthiasrohmer I'm sure there's a better way to do this

File: package.json [hunk 2]
```diff
@@ -59,6 +56,9 @@
     "signale": "1.4.0"
   },
   "devDependencies": {
+    "amp-toolbox-optimizer": "0.4.8",
+    "amp-toolbox-optimizer-express": "0.5.1",
```
Comment (matthiasrohmer) [inline package.json]: Is the middleware still a dependency?

File: platform/lib/build/pageMinifier.js (hunk 1 above)
Comment (matthiasrohmer) [reply platform/lib/build/pageMinifier.js]: None that I can think of quickly... 🤔

File: package.json (hunk 2 above)
Comment (sebastianbenz) [reply package.json]: good catch

Comment (sebastianbenz) [reply package.json]: fixed

**Pull request merged**